pkg/apiserver/rest/server.go

```diff
@@ -68,15 +74,27 @@ func createEchoServer(dbHandler databaseTypes.Database) (*echo.Echo, error) {
 	// Recover any panics into HTTP 500
 	e.Use(echomiddleware.Recover())
 
-	// Use oapi-codegen validation middleware to validate
-	// the API group against the OpenAPI schema.
-	e.Use(middleware.OapiRequestValidator(swagger))
+	// Use oapi-codegen validation middleware to validate the API group against the
+	// OpenAPI schema along with IAM provider if configured.
+	if iamEnabled {
+		iamProvider, err := iamprovider.New()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create IAM provider: %v", err)
+		}
 
-	apiImpl := &ServerImpl{
-		dbHandler: dbHandler,
+		e.Use(middleware.OapiRequestValidatorWithOptions(swagger, &middleware.Options{
+			Options: openapi3filter.Options{
+				AuthenticationFunc: iam.OapiFilterForProvider(iamProvider),
+			},
+		}))
+	} else {
+		e.Use(middleware.OapiRequestValidator(swagger))
```

Comment:
I think we need the validator middleware in anycase (also when iamEnabled)

Reply:
This is already the case:
* when IAM is enabled, we use `middleware.OapiRequestValidatorWithOptions` which contains validator with authenticator
* when IAM is disabled, we use `middleware.OapiRequestValidator` which only contains validator (and no authenticator)

Reply:
Also, `middleware.OapiRequestValidator(swagger)` is just facade call to `OapiRequestValidatorWithOptions(swagger, nil)`

Reply:
Ah ok sorry for that